src/pages/TransactionHistory/DownloadButton.tsx

```diff
@@ -68,55 +64,57 @@ export const DownloadButton = ({ txIds }: { txIds: TxId[] }) => {
 
   const generateCSV = () => {
     setIsLoading(true)
-    let report: ReportRow[] = []
-    for (let index = 0; index < txIds.length; index++) {
-      const txId = txIds[index]
-      const transaction = allTxs[txId]
-      const standardTx = getStandardTx(transaction)
-      const TxType = isSupportedContract(transaction)
-        ? TransferType.Contract
-        : standardTx?.type ?? transaction.trade?.type ?? ''
-      const buyTransfer = getBuyTransfer(transaction)
-      const sellTransfer = getSellTransfer(transaction)
-      const feeAsset = assets.find(asset => asset.assetId === transaction.fee?.assetId)
-      const input = standardTx ?? sellTransfer ?? null
-      const inputAssetId = input?.assetId ?? null
-      const output = standardTx ?? buyTransfer ?? null
-      const outputAssetId = output?.assetId ?? null
-      const inputAsset = inputAssetId ? assets.find(asset => asset.assetId === inputAssetId) : null
-      const outputAsset = outputAssetId
-        ? assets.find(asset => asset.assetId === outputAssetId)
-        : null
+
+    const report: ReportRow[] = []
+    for (const txId of txIds) {
+      const tx = allTxs[txId]
+
+      const transfers = tx.transfers.map(t => ({
+        ...t,
+        asset: assets[t.assetId],
+        marketData: defaultMarketData,
+      }))
+
+      const type = getTxType(tx, transfers)
+      const send =
+        transfers.length === 1
+          ? transfers[0]
+          : transfers.find(transfer => transfer.type === TransferType.Send)
+      const receive =
+        transfers.length === 1
+          ? transfers[0]
+          : transfers.find(transfer => transfer.type === TransferType.Receive)
+      const feeAsset = tx.fee ? assets[tx.fee?.assetId] : undefined
+
+      const typeLabel = (() => {
+        if (type === 'unknown') return 'transactionRow.unknown'
+        if (tx.data?.method) return `transactionRow.parser.${tx.data.parser}.${tx.data.method}`
```

Comment:
We will need to add `staking` and `ibc` translation `transaction.parser` entries